nautobot/extras/models/groups.py

```diff
@@ -0,0 +1,313 @@
+
+        return qs
+
+    @property
+    def members(self):
+        return self.get_queryset()
+
+    @property
+    def count(self):
+        """Return the number of objects in the group."""
+        return self.get_queryset().count()
+
+    @property
+    def map(self):
+        """
+        Accessor to automatically generated `BaseDynamicGroupMap` class.
+
+        This class object is cached on the instance after the first time it is accessed.
+        """
+
+        if getattr(self, "_map", None) is None:
+            try:
+                model = self.content_type.model_class()
+                dynamicgroupmap_class = dynamicgroup_map_factory(model)
+            except models.ObjectDoesNotExist:
+                dynamicgroupmap_class = None
+
+            self._map = dynamicgroupmap_class
+
+        return self._map
+
+    def get_absolute_url(self):
+        return reverse("extras:dynamicgroup", kwargs={"slug": self.slug})
+
+    def get_group_members_url(self):
+        """Get URL to group members."""
+        base_url = self.map.base_url
+        filter_str = self.map.urlencode(self.filter)
+
+        if filter_str is not None:
+            base_url += f"?{filter_str}"
+
+        return base_url
+
+    def get_filter_fields(self):
+        # Do not present the list of filter options until the object has been created and has a map
+        # class.
+        if not self.map:
+            return {}
+
+        if not self.present_in_database:
+            return {}
+
+        # Add all fields defined in the DynamicGroupMap to the form and populate the default value
+        fields = {}
+        for field_name, field in self.map.fields().items():
+            if field_name in self.filter:
+                field.initial = self.filter[field_name]
+
+            fields[field_name] = field
+
+        return fields
+
+    def set_filter(self, form):
+        """
+        Extract all data from `form` fields into `filter` dictionary and call `save()`.
+
+        This is called from `DynamicGroupForm.save()`.
+
+        :param form:
+            A validated instance of `DynamicGroupForm`
+        """
+        filter_fields = self.get_filter_fields()
+
+        # Populate the filterset from the incoming form's cleaned_data.
+        filterset_class = self.map.filterset_class
+        filterset_class.form_prefix = "filter"
+        filterset = filterset_class(form.cleaned_data)
+
+        # Use the auto-generated filterset form perform creation of the
+        # filter dictionary that will be saved on the instance.
+        filterset_form = filterset.form
+        filterset_form_fields = list(filterset_form.fields)
+
+        # Keep only the fields we desire for the filter.
+        for extra_field_name in filterset_form_fields:
+            if extra_field_name not in filter_fields:
+                print(f"Deleting {extra_field_name} from extra_form")
+                del filterset_form.fields[extra_field_name]
```

Comment:
I'm more than a bit confused by what's going on here. We pull the cleaned_data from the DynamicGroupForm, then shove it into a filterset (ok, makes sense so far), then render that back to a FilterForm (why?) then delete some fields from the FilterForm (shouldn't these fields have been excluded from the DynamicGroupForm in the first place? Or are we actually discarding user-provided data here?)... and finally render the filterset_form cleaned_data back to a new dict?  Why so many transformations of the data?